src/scanner.c

```diff
@@ -81,12 +93,25 @@ bool tree_sitter_scss_external_scanner_scan(void *payload, TSLexer *lexer, const
                 return false;
             }
             lexer->mark_end(lexer);
-            // We need a { to be a pseudo class selector, a ; indicates a property
-            while (lexer->lookahead != ';' && lexer->lookahead != '}' && !lexer->eof(lexer)) {
+            // We need a { to be a pseudo class selector, a ; indicates a
+            // property
+            while (lexer->lookahead != ';' && lexer->lookahead != '}' &&
```

Comment:
This is where the actual fix begins; here I initialize a boolean to track the prior character, record it if necessary, and ignore the problematic `{` if `possible_interpolation` is still true.  I then remove the `possible_interpolation` if the follow-on character was not a problematic `{`.

Man, I hate going back and writing C again.